executor/plan_replayer.go

```diff
@@ -180,33 +180,76 @@ func (e *PlanReplayerExec) Next(ctx context.Context, req *chunk.Chunk) error {
 	return nil
 }
 
-func (e *PlanReplayerExec) createFile(path string) error {
-	// Create path
+func (e *PlanReplayerExec) createFile() error {
+	var err error
+	e.DumpInfo.File, e.DumpInfo.FileName, err = GeneratePlanReplayerFile()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// GeneratePlanReplayerFile generates plan replayer file
+func GeneratePlanReplayerFile() (*os.File, string, error) {
+	path := domain.GetPlanReplayerDirName()
 	err := os.MkdirAll(path, os.ModePerm)
 	if err != nil {
-		return errors.AddStack(err)
+		return nil, "", errors.AddStack(err)
+	}
+	fileName, err := generatePlanReplayerFileName()
+	if err != nil {
+		return nil, "", errors.AddStack(err)
 	}
+	zf, err := os.Create(filepath.Join(path, fileName))
+	if err != nil {
+		return nil, "", errors.AddStack(err)
+	}
+	return zf, fileName, err
+}
 
+func generatePlanReplayerFileName() (string, error) {
 	// Generate key and create zip file
 	time := time.Now().UnixNano()
 	b := make([]byte, 16)
 	//nolint: gosec
-	_, err = rand.Read(b)
+	_, err := rand.Read(b)
 	if err != nil {
-		return err
+		return "", err
 	}
 	key := base64.URLEncoding.EncodeToString(b)
-	fileName := fmt.Sprintf("replayer_%v_%v.zip", key, time)
-	zf, err := os.Create(filepath.Join(path, fileName))
+	return fmt.Sprintf("replayer_%v_%v.zip", key, time), nil
+}
+
+func (e *PlanReplayerDumpInfo) dump(ctx context.Context) (err error) {
+	fileName := e.FileName
+	zf := e.File
+	task := &PlanReplayerDumpTask{
+		FileName:    fileName,
+		Zf:          zf,
+		SessionVars: e.ctx.GetSessionVars(),
+		TblStats:    nil,
```

Comment:
Where do we init `PlanReplayerDumpTask.TblStats`?

Reply:
This will be used in future to support https://github.com/pingcap/tidb/issues/38779